### PR TITLE
Tpetra: Fix issue in `transferAndFillComplete`

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/tpetra 

## Motivation
Proposed fix for #8447

### CODE REVIEW

**Review by jhux2 — DISMISSED** [review]
I'm jhux2, and I approve this PR.

**Comment by jhux2** [comment]
@cgcgcg As we discussed, you should add a test for this.

File: packages/tpetra/core/test/CrsMatrix/Bug8447.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    rcp (new MapType (INVALID, ArrayView<GO> (globalIDs), indexBase, comm));
+  RCP<const MapType> rowMap2 =
+    rcp (new MapType (INVALID, ArrayView<GO> (globalIDs2), indexBase, comm));
+  TEUCHOS_TEST_FOR_EXCEPTION(
+    ! rowMap->isOneToOne (), std::logic_error,
+    "In this test, the row Map is supposed to be one to one.");
+
+  RCP<CrsMatrixType> matrix = rcp (new CrsMatrixType (rowMap, 2));
+  for (size_t i = 0; i < static_cast<size_t> (globalIDs.size ()); ++i) {
+    matrix->insertGlobalValues (globalIDs[i],
```
Comment (jhux2) [inline packages/tpetra/core/test/CrsMatrix/Bug8447.cpp:127]: @cgcgcg This adds to the UVM backlog.  @trilinos/tpetra @csiefer2 What's the canonical way to do this without UVM?

Comment (cgcgcg) [reply packages/tpetra/core/test/CrsMatrix/Bug8447.cpp:127]: Ugh, indeed. Lemme rewrite this with views.

**Review by jhux2 — APPROVED** [review]
Sehr gut!

**Pull request merged**